packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -85,6 +100,95 @@ class SemanticsTag {
   String toString() => '${objectRuntimeType(this, 'SemanticsTag')}($name)';
 }
 
+/// The result that contains the arrangement for the child
+/// [SemanticsConfiguration]s.
+///
+/// When the [PipelineOwner] builds the semantics tree, it uses the returned
+/// [ChildSemanticsConfigsResult] from
+/// [SemanticsConfiguration.childConfigsDelegate] to decide how semantics nodes
+/// should form.
+///
+/// Consider using [ChildSemanticsConfigsResultBuilder] to build the result.
+abstract class ChildSemanticsConfigsResult {
```

Comment:
Why split this in an abstract and a private class? Can this just be a concrete class with a private constructor that's only used by the builder?